x-pack/legacy/plugins/maps/public/components/map/mb/view.js

```diff
@@ -127,13 +127,31 @@ export class MBMapContainer extends React.Component {
   }, 256);
 
   _getIdsForFeatures(mbFeatures) {
-    return mbFeatures.map((mbFeature) => {
+    const uniqueFeatures = [];
```

Comment:
Use a Map instead of an array.

Reply:
Discussed offline. Map would require a unique key that would be concatenation of feature Id and layer Id. This would use more memory than the current implementation.